### DISCUSSION TITLE
XAML Preview not updating

### DISCUSSION THREAD

**Discussion by astromich** [AvaloniaUI/Avalonia#16437] [General]
Hello all

I'm new to Avalonia, only downloaded and installed it yesterday. I have the problem that the XAML preview is not updating when I make changes to the code. However if I save the file, close and then open it again, my changes are reflected in the preview. It's a bit cumbersome to work like this. Google didn't really help unfortunately. Is this know, or is there a fix to this?
I'm using the latest version of Visual Studio Community 2022 and installed Avalonia thru the available vsix installer.

Thanks
Martin

**Comment by stevemonaco** [comment]
It seems like there was a bug introduced somewhere in the build process that didn't manifest though beta/RC testing until the AvaloniaVS extension was updated.

#16427 seems to fix it for me, but you need to locally setup the [PR package feed](https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed#all-builds-feed) to use it. I'm not sure when the team plans to ship this fix, assuming it's enough to fix the issue. This should also affect VS Code, but Rider appears to be unaffected in my quick testing.

**Comment by workgroupengineering** [comment]
While waiting for the fix to be released, you can try adding the following target to your `Project.Desktop.csproj`.

```xml
  <Target Name="UpdateAvaloniaReferences" AfterTargets="CompileAvaloniaXaml">
    <PropertyGroup>
      <AvaloniaXamlReferencesTemporaryFilePath Condition="'$(AvaloniaXamlReferencesTemporaryFilePath)' == ''">$(IntermediateOutputPath)/Avalonia/references</AvaloniaXamlReferencesTemporaryFilePath>
    </PropertyGroup>
    <WriteLinesToFile File="$(AvaloniaXamlReferencesTemporaryFilePath)" Lines="@(ReferencePathWithRefAssemblies)" Overwrite="true" />
  </Target>
```

**Comment by astromich** [comment]
Thanks for the suggestions. Unfortunately both solutions do not work for me, the view stays unaffected when making changes, saving of compiling. I have to reopen the file for the changes to be applied in the preview. I guess I'll just have to wait for an official fix then.

**Comment by astromich** [comment]
Interesting. After creating a new project and copying the few changes that I made into it, the view is updating as it would be expected. Not sure what I did differently this time, I tried with a new project multiple times yesterday before posting, and nothing seemed to work.
